crates/driver/src/infra/blockchain/mod.rs

```diff
@@ -224,6 +226,17 @@ impl Ethereum {
             })
             .map_err(Into::into)
     }
+
+    async fn simulation_gas_price(&self) -> Option<eth::U256> {
```

Comment:
Why don't we use the existing `gas_price` method?

Reply:
Completely missed that function. Still kept the new private function to have a single place to document why we simulate with a gas price in the first place but it's now calling the other function.

Reply:
If it just me feel free to keep it, but I find it quite intuitive why we should set a gas price in the simulation (to make sure the code behaves as it will later on chain). But no strong feeling.